pkg/operation/botanist/vpnshoot.go

```diff
@@ -0,0 +1,89 @@
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnshoot"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// DefaultCoreDNS returns a deployer for the VPNShoot
+func (b *Botanist) DefaultVPNShoot() (vpnshoot.Interface, error) {
+	var (
+		image                  *imagevector.Image
+		err                    error
+		nodeNetwork            = b.Shoot.GetInfo().Spec.Networking.Nodes
+		ReversedVPNHeader      string
+		ReversedVPNEndPoint    string
+		ReversedVPNOpenVPNPort string
+		NodeNetworkCIDR        string
+	)
+	if nodeNetwork != nil {
+		NodeNetworkCIDR = *nodeNetwork
+	}
+	if b.Shoot.ReversedVPNEnabled {
+		image, err = b.ImageVector.FindImage(charts.ImageNameVpnShootClient, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+		if err != nil {
+			return nil, err
+		}
+		ReversedVPNHeader = "outbound|1194||" + vpnseedserver.ServiceName + "." + b.Shoot.SeedNamespace + ".svc.cluster.local"
+		ReversedVPNEndPoint = b.outOfClusterAPIServerFQDN()
+		ReversedVPNOpenVPNPort = "8132"
+	} else {
+		image, err = b.ImageVector.FindImage(charts.ImageNameVpnShoot, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	values := vpnshoot.Values{
+		Image:                  image.String(),
+		VPAEnabled:             b.Shoot.WantsVerticalPodAutoscaler,
+		ReversedVPNEnabled:     b.Shoot.ReversedVPNEnabled,
+		PodNetworkCIDR:         b.Shoot.Networks.Pods.String(),
+		ServiceNetworkCIDR:     b.Shoot.Networks.Services.String(),
+		NodeNetworkCIDR:        NodeNetworkCIDR,
+		ReversedVPNHeader:      ReversedVPNHeader,
+		ReversedVPNEndPoint:    ReversedVPNEndPoint,
+		ReversedVPNOpenVPNPort: ReversedVPNOpenVPNPort,
+	}
+
+	return vpnshoot.New(
+		b.K8sSeedClient.Client(),
+		b.Shoot.SeedNamespace,
+		values,
+	), nil
+}
+
+func (b *Botanist) DeployVPNShoot(ctx context.Context) error {
+	if b.Shoot.ReversedVPNEnabled {
+		b.Shoot.Components.SystemComponents.VPNShoot.SetSecrets(vpnshoot.Secrets{
+			TLSAuth: component.Secret{Name: vpnshoot.SecretNameTLSAuth, Checksum: b.LoadCheckSum(vpnshoot.SecretNameTLSAuth), Data: b.LoadSecret(vpnshoot.SecretNameTLSAuth).Data},
+			Server:  component.Secret{Name: vpnshoot.VpnShootSecretName, Checksum: b.LoadCheckSum(vpnshoot.VpnShootSecretName), Data: b.LoadSecret(vpnshoot.VpnShootSecretName).Data},
+		})
+	} else {
+		b.Shoot.Components.SystemComponents.VPNShoot.SetSecrets(vpnshoot.Secrets{
+			TLSAuth: component.Secret{Name: vpnshoot.SecretNameTLSAuth, Checksum: b.LoadCheckSum(vpnshoot.SecretNameTLSAuth), Data: b.LoadSecret(vpnshoot.SecretNameTLSAuth).Data},
+			DH:      component.Secret{Name: vpnshoot.SecretNameDH, Checksum: b.LoadCheckSum(vpnshoot.SecretNameDH), Data: b.LoadSecret(vpnshoot.SecretNameDH).Data},
+			Server:  component.Secret{Name: "vpn-shoot", Checksum: b.LoadCheckSum("vpn-shoot"), Data: b.LoadSecret("vpn-shoot").Data},
```

Comment:
We should probably introduce a constant here as well. `VpnShootSecretName` is current `vpn-shoot-client`, so maybe we rename it to `SecretNameVPNShootClient` and can then introduce `SecretNameVPNShoot = "vpn-shoot".